state/aws/dynamodb/dynamodb.go

```diff
@@ -250,3 +293,24 @@ func (d *StateStore) marshalToString(v interface{}) (string, error) {
 
 	return jsoniterator.ConfigFastest.MarshalToString(v)
 }
+
+//Parse and process ttlInSeconds
+func (d *StateStore) parseTTL(req *state.SetRequest) (*int64, error) {
+	if val, ok := req.Metadata["ttlInSeconds"]; ok && val != "" {
+		parsedVal, err := strconv.ParseInt(val, 10, 0)
+		if err != nil {
+			return nil, err
+		}
+		parsedInt := int64(parsedVal)
+		//To follow the spec
+		if parsedInt == -1 {
```

Comment:
`-1` was intended to be used to explicitly persist a record, for example in the case where the state store supports a default TTL that is applied to every new entity.

It does not seem that DynamoDB supports such a concept. Perhaps you should return for every negative value?

Reply:
What I tried to archive;
Imagine you create the following item:
```
key: dummyKey
value: someValue
ttl: 90
```

And 20 seconds later you do:
```
key: dummyKey
value: someValue
```

Will this override the existing TTL field so that DynamoDB won't delete it after 70 seconds?

Reply:
That depends on how Dynamo DB handles this (which I personally don't know as I am not a user of DynamoDB). In general I think that should indeed reset the TTL to whatever the default is (no TTL?).

If there is a default TTL (which I think Dynamo DB doesn't have) you might need to specify explicitly that you want to ignore the default TTL and persist the item. That's what `-1` was intended to be for.

I think you don't want have to worry about that -- I can't seem to find anything about default TTLs / globally defined TTLs in Dynamo DB, so it probably doesn't apply here.

Reply:
Ok, I removed the IF statement and modified the tests a bit.